adapters/kafka/kafka.go

```diff
@@ -0,0 +1,129 @@
+package kafka
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gliderlabs/logspout/router"
+	"gopkg.in/Shopify/sarama.v1"
+)
+
+func init() {
+	router.AdapterFactories.Register(NewKafkaAdapter, "kafka")
+}
+
+type KafkaAdapter struct {
+	route    *router.Route
+	brokers  []string
+	topic    string
+	config   *sarama.Config
+	client   sarama.Client
+	producer sarama.AsyncProducer
+	tmpl     *template.Template
+}
+
+func NewKafkaAdapter(route *router.Route) (router.LogAdapter, error) {
+	brokers, topic, err := parseKafkaAddress(route.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	var tmpl *template.Template
+	if text := os.Getenv("KAFKA_FORMAT"); text != "" {
+		tmpl, err = template.New("kafka").Parse(text)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't create template from KAFKA_FORMAT: %v", err)
+		}
+	}
+
+	config := buildConfig(route.Options)
+	client, err := sarama.NewClient(brokers, config)
```

Comment:
fyi, you no longer need to create/manage a `Client` at all unless you need it for something specific, you can simply call `NewAsyncProducer(brokers, config)`